azurelinuxagent/common/protocol/wire.py

```diff
@@ -780,45 +779,40 @@ def update_host_plugin_from_goal_state(self):
         goal_state = GoalState(self)
         self._update_host_plugin(goal_state.container_id, goal_state.role_config_name)
 
-    def update_goal_state(self, force_update=False, is_retry=False):
+    def update_goal_state(self, force_update=False):
         """
         Updates the goal state if the incarnation or etag changed or if 'force_update' is True
         """
         try:
             #
-            # The entire goal state needs to be retrieved from the WireServer (via the GoalState class), and the HostGAPlugin
+            # The goal state needs to be retrieved using both the WireServer (via the GoalState class) and the HostGAPlugin
             # (via the self._fetch_vm_settings_goal_state method).
             #
-            # We fetch it in 3 parts:
+            # We always need at least 2 queries: one to the WireServer (to check for incarnation changes) and one to the HostGAPlugin
```

Comment:
> one to the WireServer (to check for incarnation changes)

I was under the impression that this was going to be abstracted by the HostGA Plugin? Is the vmSettings only going to be an abstraction for the `ExtensionConfig` part of the GoalState and not the complete goal state?

Reply:
The vmsettings don't include the incarnation number, that comes from the wireserver. Even if Fast Track is enabled and supported we also need to check for other changes in the goal state (e.g JIT access would change the remote users). Will add a little more text to this comment

Reply:
vmSettings would still catch the JIT user scenario right? Since it is fetching the goal states from both channels and comparing and giving us the latest one right?

Also, does HostGA plan to move the complete goal state to vmSettings (certs, roleConfig, etc) or is it going to be limited to the ExtensionsConfig?

Reply:
vmsettings is not a complete goal state as of today, so it would not include remote users

afaik, vmsettings will not go beyond extensions (i.e. no remote users, for example). as of today, the certs are still missing, but i believe they will be added in a (as yet unspecified) future version